nasa/earthdata-search
# EDSC-4002: added retrieveVariablesRequest and added it to focusedCollection.js and project.js

# Overview

### What is the feature?

Allow for variables to be fully returned for collection requests.

### What is the Solution?

Added the function retrieveVariablesRequest and added it to focusedCollections.js and project.js
which goes through and returns all associated variables for collections.

### What areas of the application does this impact?

List impacted areas.

# Testing

### Reproduction steps

- **Environment for testing:**UAT
- **Collection to test with:**C1241484650-LPCLOUDUAT

1. navigate to localhost:8080/search?ee=uat
2. search for C1241484650-LPCLOUDUAT
3. Click on the collection
4. Ensure that everything loads properly and if possible, use Redux chrome extension to make sure all 4000+ variables are added when the UPDATE_COLLECTION_METADATA is dispatched.
5. add the collection to the project and click Download
6. Ensure that everything loads properly and if possible, use Redux chrome extension to make sure all 4000+ variables are added when the UPDATE_COLLECTION_METADATA is dispatched.

### Attachments

Please include relevant screenshots or files that would be helpful in reviewing and verifying this change.
*Nothing changes on the frontend

# Checklist
